engine/render/src/render/constant.cpp

```diff
@@ -102,9 +102,10 @@ struct NamedConstantBuffer
 {
     struct Constant
     {
-        dmhash_t   m_NameHash;
-        uint32_t   m_ValueIndex;
-        uint32_t   m_NumValues;
+        dmhash_t                                m_NameHash;
+        uint32_t                                m_ValueIndex;
+        uint32_t                                m_NumValues;
+        dmRenderDDF::MaterialDesc::ConstantType m_Type;
```

Comment:
We could potentially skip this member since the actual render constant should have this information already, but it's a bunch of extra code to fetch it.

Reply:
Actually we need to have access to what type the value was created with originally since a constant buffer can be shared across different materials and doesn't have the type information available when pushing values to the lua stack in the _index function for example..